object-construction-qual/build.gradle

```diff
@@ -0,0 +1,41 @@
+plugins {
+    // Apply the java plugin to add support for Java
+    id 'java'
+
+    // For bringing in AWS dependencies. We're using the "legacy" version
+    // listed here: https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-project-gradle.html,
+    // because the build fails when we use the Gradle 5+ way, saying that it can't find ec2.
+    // I believe that this is because we need the AWS SDK during testing, but not during the build.
+    id 'io.spring.dependency-management' version '1.0.7.RELEASE'
+
+    id "com.diffplug.gradle.spotless" version "3.23.0"
+}
+
+sourceCompatibility = 1.8
+
+dependencies {
+    // This dependency is found on compile classpath of this component and consumers.
+    implementation 'org.checkerframework:checker:2.9.0'
```

Comment:
This should also be `checker-qual`, because we don't want anyone who depends on the annotations to get the framework on their compile classpath.

I would also add a dependency on the annotation package for the returns receiver checker here, so that `@This` is on the classpath automatically without the need for a second annotation dependency (this is the current behavior).